language-server/src/server.js

```diff
@@ -135,6 +144,40 @@ connection.listen();
 
 const documents = new TextDocuments(TextDocument);
 
+// CONFIGURATION
+
+const documentSettings = new Map();
+let globalSettings = {};
+
+async function getDocumentSettings(resource) {
+  if (!hasConfigurationCapability) {
+    return globalSettings;
+  }
+
+  let result = documentSettings.get(resource);
+  if (!result) {
+    result = await connection.workspace.getConfiguration({
+      scopeUri: resource,
+      section: "jsonSchemaLanguageServer"
+    });
+    if (result !== null) {
+      documentSettings.set(resource, result);
+    }
+  }
+
+  return result ?? {};
```

Comment:
Actually, I was think more like applying the default before adding it to the Map. This feels more straightforward.

```suggestion
  if (!documentSettings.has(resource)) {
    const result = await connection.workspace.getConfiguration({
      scopeUri: resource,
      section: "jsonSchemaLanguageServer"
    });
    documentSettings.set(resource, result ?? globalSettings);
  }

  return documentSettings.get(resource);
```